### ISSUE TITLE
No notifications in Android app

### ISSUE THREAD

**Issue by MarcS1975** [nextcloud/talk-android#1498]
## Steps to reproduce
1. sending talk message from User 1 to User 2
2. No notification on User 2's phone

### Expected behaviour
User 2 should get pop up notification

### Actual behaviour
nothing happens

## Device information

**Device:** Fairphone2/…
Samsung S10

**Android version:** 10/…
Amdroid 10

**Talk version:** 11.0.0
12.0.1

## Server information
apache 2.4

**Nextcloud version:** (see admin overview page: `/index.php/settings/admin/overview`)
 Nextcloud server 22.0.0

**Talk version:** (see apps admin page: `/index.php/settings/apps`)
12.0.1

**Custom Signaling server configured:** yes/no and version (see talk admin settings: `/index.php/index.php/settings/admin/talk#signaling_server`)
no
**Custom TURN server configured:** yes/no (see talk admin settings: `/index.php/settings/admin/talk#turn_server`)
no
**Custom STUN server configured:** yes/no (see talk admin settings: `/index.php/settings/admin/talk#stun_server`)
no
### Server log (data/nextcloud.log)
<details>

```
nextcloud log attached
```
[nextcloud.log](https://github.com/nextcloud/talk-android/files/6885826/nextcloud.log)

</details>

**Comment by bugbountyguy** [comment]
Just wanted to plus one this. My android talk app has stopped sending notifications to my phone. I do daily updates on the server so I'm not sure if that's related, or if it's the android app after a recent release, but something definitely busted.
Android 11, Nextcloud running in Docker using apache and latest image available

**Comment by penguin86** [comment]
Just to say that notifications seems to depend on the Google Services framework, thus not working on AOSP or non-google devices.
I found this notification app stub [here](https://github.com/Andrewerr/NextcloudServices), but ATM the notifications are not clickable, I'm trying to figure out the simplest way to generate an Intent to open the relevant apps.

**Comment by bugbountyguy** [comment]
I do have the Google play services on mine and working and haven't made any changes to that recently. Notifications for talk has been working on my phone for the past year or two since I first used it. The only thing I can think of is the Google framework was a recent addition and somehow isn't working. But other apps on my phone that also depend on it work fine with notifications. 
Interestingly one of my chats did give me notifications today, but not the others I expected any from. Seems to be selective or something.

**Comment by MarcS1975** [comment]
Yes - i also have notifications failing 80% of users/chats. But then some random users work sometimes.
I have not changed anything in my Google Service framework and all other chat apps deliver notifications 100% reliably.

**Comment by bgbearcatfan** [comment]
I replied at the previous thread as well (https://help.nextcloud.com/t/nextcloud-talk-push-notifications-not-working/120873/6).  But just want to chime in saying I'm having the same issues as well.  NC 22.1.0, android talk 12.1.2, using Pixel 4XL.

**Comment by Bl4DEx** [comment]
Hello,

I also want to report my test results:

Nextcloud Server: 21.0.3
Nextcloud Talk: 11.3.1

Only two users in the nextcloud network:
Talk iOS (12.1.0)
Talk Android (12.1.2) (Xiaomi Mi 9T Pro; Android 11; Power saving Mode disabled for Talk)

Talk iOS works perfectly fine.
Talk Android is having problems with the push notifications. It seems to work most of the time but if Talk is closed for a while (some hours) , there are no notifications. Re-open the app does not enable the notifications. The only workaroud for me is to go to the Android settings of Talk and disable and enable the push notifications.
It works fine afterwards.

I was using Talk Android 11.x for several months and never experienced this behaviour with this phone.

Is there any useful testing strategy I could do to help find the problem?

**Comment by MarcS1975** [comment]
I don't believe thst it has anything to do with Google play. All other chat
apps work perfectly fine and deliver notifications. It's a shame that
nobody on the Nextcloud team feels responsible to debug this. There must be
someone who has written the code and the notifications have been buggy for
years although it's such an essential function for a chat app!

**Comment by Valdnet** [comment]
I confirm that PUSH notifications on Android 10 are also not always received.

App:
Nextcloud 22.0.0
Talk 12.0.1
Talk Android 12.1.2

**Comment by bgbearcatfan** [comment]
Android talk app 12.2.0 has the same issue.

**Comment by dmanea28** [comment]
Same issue for me for Talk version 12.2.1

**Comment by MarcS1975** [comment]
Thanks for the tip. In my case, the performance has suddenly Gotten even
worse. In the past the apple app users got notifications but since about 3
weeks they also stopped getting them. We have not changed anything on our
server or the client phones. It's a shame that NC is not supporting
notifications and nobody seems to take this bug seriously. A chat app
without notifications is just useless and will never go anywhere...

**Comment by MarcS1975** [comment]
I have done similar resets and reinstalls with Samsung S9, S10, iPhone
10,11 and 12. The bug still persists.

**Comment by bgbearcatfan** [comment]
Unfortunately, notifications are still sporadic. Good luck everyone.  This problem dates back to July, and I am not willing to wait around anymore.  Will be migrating to a different cloud/chat option.  Hope you all get it straightened out at some point!

**Comment by bugbountyguy** [comment]
As of about 3 nights ago, my notifications spontaneously started working on their own again. I have no idea what changed, but I run updates on my system every night and am thinking an update was in that fixed this... Maybe. I recommend anyone else having this issue to try and update their nextcloud instance to the latest version and see if it randomly starts working for you as well. Either the update fixed it or I just got lucky this time around? Lol

**Comment by MarcS1975** [comment]
Awesome, thanks for the update!
What version are you running now?

**Comment by bugbountyguy** [comment]
Just checked, looks like I'm on nextcloud 22.2.0. There's a patch version I think that Docker hasn't picked up on yet or something but that's what it looks like the last update pulled in.

**Comment by MarcS1975** [comment]
I'm on the same but still no notifications.

**Comment by bugbountyguy** [comment]
Well, damn :( was hoping the new minor version had something. I'll keep an eye out for if they stop working again lol. But for the past couple of days I've been good... I'll dig deeper into my logs later too and see if I can pinpoint any changes that might help.

**Comment by MarcS1975** [comment]
Sounds good, thanks!
It is becoming clear to me that the NC developers are really beginners and
all learning. As soon as one has some knowledge they get hired into real
companies in California. So what they are left with is a bunch of beginners
that don't known what the code does.

**Comment by bugbountyguy** [comment]
I dug around the logs and came up short. I checked all my Nextcloud logs (filtering out common deprecated noise and whatnot) for the past 2 weeks. There was nothing different or odd that stood out to me, personally. 

Here are some things that _might_ be worth comparing with what I have running, but may not necessarily mean anything:
```
PHP version 8.0.12
mysql version 10.5.9
Apache info:
  Server version: Apache/2.4.51 (Debian)
  Server built:   2021-10-07T17:49:44

Nextcloud Version: Nextcloud 22.2.1
Nextcloud Talk Version: 12.1.2	
Android Talk Version: 12.2.1
```
It appears there was also a docker-compose update on my system... Not sure if that's even relevant, but it did happen around the time notifications began working. It was version 1.29.2-r1. Lastly, I also had a firewall update so if you're behind a custom firewall (pfsense, opnsense, ipfire, etc.) maybe update that and see if it helps. Again, doubtful that had anything to do with it, but doesn't hurt to update anyway. 

That all being said, I'm still getting all my notifications. I did no other Android software/system updates recently (those are manual instead of automatic), or at least not in the past week. 

If none of this info helps, then I'm really not sure what it could have been. :( Just got lucky I guess.

**Comment by w202mg** [comment]
Same issue here. I noticed that I do not get the notifications if I'm logged in with 2 accounts on my phone. once I delete the 2dn one and have only one left it works well. don't know if I have to open a separate issue or not.

**Comment by bugbountyguy** [comment]
Just like how they spontaneously started working, they have no spontaneously stopped working again, LOL. Back to square one for me, too.

**Comment by jbrockc** [comment]
I upgraded my server from Debian 10 to Debian 11, and now my notifications are working again. I did not change any software on my phone which runs Lineage OS. My wife has a Pixel phone with the stock OS, and her notifications have been working all along. On my server, I am running the snap package.

**Comment by bugbountyguy** [comment]
> I upgraded my server from Debian 10 to Debian 11, and now my notifications are working again. I did not change any software on my phone which runs Lineage OS. My wife has a Pixel phone with the stock OS, and her notifications have been working all along. On my server, I am running the snap package.

My notifications have also been working the past few weeks too. About a week or so after the last post I made, they started working again and have been since then... Not sure if there was an update to the docker container that Nextcloud provides, or the newest Nextcloud finally fixed it (Talk app also updated on the phone, so maybe that?).

**Comment by Mer0me** [comment]
Nextcloud Hub II (23.0.0)
NC App Talk 13.0.1
Android App 13.0.0 on a e/os/ Fairphone 

Notifications are not working at all on my phone and are randomly working on the browser (Firefox 95.0.1 / Xubuntu). They work well on the same conversation on an ios device.

**Comment by Mer0me** [comment]
Thanks to this [reddit thread](https://www.reddit.com/r/NextCloud/comments/m85z3c/nextcloud_talk_android_how_to_enable_audible/) I finally get notifications on my (fair)phone without Google Services by downloading the APK on apkpure website. The F-Droid package seems to be incomplete and notifications didn't work with it.
I can now hear and see call notifications and other text message notifications on the phone.

**Comment by mahibi** [comment]
@Mer0me please note that the F-droid package doesn't contain Google Play services so Notifications won't work.

To check why notifications don't work please always use this checklist:
https://github.com/nextcloud/talk-android/blob/master/docs/notifications.md
(anyone who can contribute knowledge to the list is welcome to do so)

Furthermore, please update to spreed 12.2.2, as the fix https://github.com/nextcloud/spreed/pull/6300 should make the notifications more reliable. 
Anyone who is on spreed >=12.2.2 and made sure he followed the checklist and still has problems with notifications, feel free to let us know in this thread.

**Comment by apady** [comment]
Nextcloud Hub II (23.0.0)
NC App Talk 13.0.1
Android App 13.0.0 

When launching  **Talk(For Android) APP**(Nextcloud (For Android) APP  is  alike) for the first time, it will display a small message on the screen,saying "**the system push notification was disabled**", and then I can not get any **system push notification** when others send message to me, neither chat nor mobile call message, unless keep the Talk APP running on the top of screen.

**Comment by bugbountyguy** [comment]
Welp, this appears to be busted again lol. I had to re-install Talk on my phone today and that resulted in me getting the latest version available (v13.0.0). And yes, I got this straight from the google play store.

When running the `notification:test-push` command, it says "No devices found for user". Like apady mentioned above, immediately after logging in, I see "the system push notification was disabled" as well. I decided to try and manually to install the old 12.2.1 version (like my wife's phone has, where notifications work) but I get an error saying my server's version of Talk is too old (It's actually 13.0.4, so... it's newer). Then I get no notifications and the same error when doing test-push.

Any fixes on this by chance? Even if it means sitting this thing up as a service in the background (which I actively do for several apps and have zero issues with battery)... Whatever can get this to work because this is becoming quite a stressful thing to deal with :(

**Comment by JimKarvo** [comment]
Are there any updates?

There is something weird with notifications.
Yesterday morning I installed the nextcloud to new server, tried the Android push notifications and worked like a charm.

Tonight, after 36 hours of first installation, I decided to reinstall next cloud from the scratch again.
Same server, same ip, same domain name.
But this time notifications are not working.

Till now I thought that I had problem with my old next cloud installation (v21).

**Comment by bgbearcatfan** [comment]
@JimKarvo , I understand your frustration.  Luckily for myself, I found that most issues went away after i upgraded NextCloud to version 23.  

However, one thing i noticed.... If you have the nextcloud web interface opened up on a desktop/laptop, even if the focus ISN'T on the nextcloud browser tab, AND, even if the talk app is NOT opened in the web interface (for example, if you have the "files" or "calendar" app opened instead), it prevents a push notification from being sent to my android phone.  Even WITHOUT the conversation being marked as read, simply being logged into the web interface is enough to disable android push notifications for talk.

It's one of the many strange behaviors i have noticed over the last year.  I hope this helps...

**Comment by JimKarvo** [comment]
> @JimKarvo , I understand your frustration. Luckily for myself, I found that most issues went away after i upgraded NextCloud to version 23.
> 
> However, one thing i noticed.... If you have the nextcloud web interface opened up on a desktop/laptop, even if the focus ISN'T on the nextcloud browser tab, AND, even if the talk app is NOT opened in the web interface (for example, if you have the "files" or "calendar" app opened instead), it prevents a push notification from being sent to my android phone. Even WITHOUT the conversation being marked as read, simply being logged into the web interface is enough to disable android push notifications for talk.
> 
> It's one of the many strange behaviors i have noticed over the last year. I hope this helps...

Cause I spend a lot of hours to check what goes wrong even with new installation, the problem exists only if user selected another language than English.

**Comment by TommySchwartz** [comment]
Where do you have to set the language to english? Talk app, nextcloud profile, nextcloud server, phone system, phone keyboard?

Brittish english, US english?

Can't get it to work 😔

No notifications on Android since I switched to Nextcloud AIO. iOS and firefox on computers work fine.

**Comment by MarcS1975** [comment]
Yes that behaviour is known. When logged into the Web the phone
notifications stop. They called it a feature. I think it's a nuisance and i
have never come across any other platform that does that. I am not sure why
this is auch a difficult and unsolvable issue. Maybe the developers just
don't care about Android.

**Comment by MarcS1975** [comment]
Which cloud/chat option are you switching to?

**Comment by mnlipp** [comment]
I've just run into the notification problem after installing v14.0.2 from f-droid. Installing from google play "fixed" it.

**Comment by JimKarvo** [comment]
> I've just run into the notification problem after installing v14.0.2 from f-droid. Installing from google play "fixed" it.

Version from Google okay includes the libraries for push notifications. (Push notifications working via Google firebase).

**Comment by mrpete69** [comment]
I just want to add a possible reason for missed notifications. I use pihole as adblocker at home where my nextcloud instance also is located. When collector.github.com was blocked, notifications weren't sent, but unblocking that address solved my problem.

**Comment by timkrueger** [comment]
I think because of the age of this ticket the reasons why notifications a different points was not working differs.

__In general push notifications are only working if you install NC Talk for Android from the GooglePlay store.___

Please feel free to open a new issue if notifications issues still existing.

**Comment by AgentZman** [comment]
Mine is failing notifications for Samsung only. 
All androids and Mac notifications are working for users, but just not Samsung. I have been all over this phone and do not know why. 
If I actually open the Talk app on Samsung then a notification comes in. This is very strange. 
Also NO notification (1) next to the icon until after I open the Talk App. 

What could cause this. Likely Samsung related but also annoying since family new phones are samsung.

**Comment by timkrueger** [comment]
> Mine is failing notifications for Samsung only. All androids and Mac notifications are working for users, but just not Samsung. I have been all over this phone and do not know why. If I actually open the Talk app on Samsung then a notification comes in. This is very strange. Also NO notification (1) next to the icon until after I open the Talk App.
> 
> What could cause this. Likely Samsung related but also annoying since family new phones are samsung.

Did you have a look at our [notification guide](https://github.com/nextcloud/talk-android/blob/master/docs/notifications.md)? 

If that not helps, please [open a new issue](https://github.com/nextcloud/talk-android/issues/new?assignees=&labels=bug%2C0.+Needs+triage&template=bug_report.yml).

**Comment by TommySchwartz** [comment]
I only have problem with notifications for the Talk-app itself on android.

Notifications for Talk:
Desktop - Firefox macOS - Works (messages go via nextcloud notification?)
Desktop - Firefox Windows - Works (messages go via nextcloud notification?)
Desktop - Edge Windows - Works (messages go via nextcloud notification?)
Desktop - Nextcloud Windows - Works (messages go via nextcloud notification?)
Desktop - Nextcloud macOS - Works (messages go via nextcloud notification?)

Mobile - Talk iOS - Works
Mobile - Firefox LineageOS - Works (messages go via nextcloud notification?)
Mobile - Nextcloud LineageOS - Works (yes I have the gplay or google play store version)
Mobile - Talk LineageOS - NOT working (yes I have the gplay or google play store version)

As you see in the list above, even the Nextcloud app on android get talk notifications, but not the talk app itself! 

I have tried the notifications guide, same behavior. Push notifications get sent to all other devices but not to the android talk app.

This problem av been over a year! I have stopped trying different versions and trying nightly builds in hope to get this fixed. 
Can only conclude it is the talk app that is the problem.

**Comment by MarcS1975** [comment]
Same here but it’s NOT a Samsung issue. It’s a Nextcloud issue or Talk
issue because none of my other 120 android apps have any problems on
Samsung phones.

**Comment by Samael774** [comment]
Hi. No notifications on Talk 17, latest SNAP Nextcloud, Xiaomi 11T Pro with all permissions and notifications enabled for Talk. Same thing my other phone Samsung Galaxy Note10.

**Comment by MarcS1975** [comment]
Same here. No notifications showing on Samsung s20+ although everything is
enabled.

**Comment by DanielSiepmann** [comment]
https://github.com/nextcloud/talk-android/issues/1498#issuecomment-1216481003 did the trick, I installed via another store. Using Aurora Store fixed the issue. I could than head over to the notifications settings of the app within the system settings and could activate them.

I've the feeling notifications take some time until they arrive. They aren't as fast as other chat solutions. But that's fine I guess.

**Comment by faxotherapy** [comment]
Also for me too. Installed Talk from F-Droid store. Notif activated all round. Still not working on Android.

**Comment by hoshi411** [comment]
Genius idea!  Why not make notifications work by default and give the users the option to opt out if they want to?  I'm so tired of notifications not working and having to track down some obscure forum and tutorial to get the basic functionality.  NOTIFICATIONS should be the default , NOT the fringe extra feature.

**Comment by jbrockc** [comment]
I have been using ntfy for notifications using the Talk F-Droid version. I just posted a video about what I do. I hope it can be of help.
https://youtu.be/0a6PpfN5PD8?si=1igz3kB6Rk2fS7wz
